drivers/dma/dma_mcux_lpc.c

```diff
@@ -740,40 +740,40 @@ static const struct dma_mcux_lpc_config dma_##n##_config = {		\
 }
 
 #ifdef FSL_FEATURE_DMA_NUMBER_OF_CHANNELS
-#define TOTAL_DMA_CHANNELS FSL_FEATURE_DMA_NUMBER_OF_CHANNELS
+#define TOTAL_DMA_CHANNELS(idx) FSL_FEATURE_DMA_NUMBER_OF_CHANNELS
 #else
-#define TOTAL_DMA_CHANNELS FSL_FEATURE_DMA_NUMBER_OF_CHANNELSn		\
-				((DMA_Type *)DT_INST_REG_ADDR(n))
+#define TOTAL_DMA_CHANNELS(idx) FSL_FEATURE_DMA_NUMBER_OF_CHANNELSn		\
+				((DMA_Type *)DT_INST_REG_ADDR(idx))
 #endif
 
 #define DMA_INIT(n) \
-									\
-	static const struct dma_mcux_lpc_config dma_##n##_config;	\
-									\
-	static struct channel_data dma_##n##_channel_data_arr		\
-				[DT_INST_PROP(n, dma_channels)] = {0};	\
-									\
-	static struct dma_otrig dma_##n##_otrig_arr			\
-			[DT_INST_PROP_OR(n, nxp_dma_num_of_otrigs, 0)]; \
-									\
-	static int8_t							\
-		dma_##n##_channel_index_arr[TOTAL_DMA_CHANNELS] = {0};	\
-									\
-	static struct dma_mcux_lpc_dma_data dma_data_##n = {		\
-		.channel_data = dma_##n##_channel_data_arr,		\
-		.channel_index = dma_##n##_channel_index_arr,		\
-		.otrig_array = dma_##n##_otrig_arr,			\
-	};								\
-									\
-	DEVICE_DT_INST_DEFINE(n,					\
-			    &dma_mcux_lpc_init,				\
-			    NULL,					\
-			    &dma_data_##n, &dma_##n##_config,		\
-			    PRE_KERNEL_1, CONFIG_DMA_INIT_PRIORITY,	\
-			    &dma_mcux_lpc_api);				\
-									\
-	DMA_MCUX_LPC_CONFIG_FUNC(n)					\
-									\
+										\
+	static const struct dma_mcux_lpc_config dma_##n##_config;		\
+										\
+	static struct channel_data dma_##n##_channel_data_arr			\
+				[DT_INST_PROP(n, dma_channels)] = {0};		\
+										\
+	static struct dma_otrig dma_##n##_otrig_arr				\
+			[DT_INST_PROP_OR(n, nxp_dma_num_of_otrigs, 0)];		\
+										\
+	static int8_t								\
+		dma_##n##_channel_index_arr[TOTAL_DMA_CHANNELS(n)] = {0};	\
```

Comment:
I do not understand this change. You are indirectly passing `n`

Reply:
@EmilioCBen can you take another look at this change? It seems to be causing a build failure in the UART async API test.

Reply:
@danieldegrasse that build failure was because the uart node needed its dmas phandle property, which I think emilio has fixed

but I also do not like this SDK-based macro, I see no reason not to use DT_INST_PROP_OR to get the number of channels like the array two lines above uses for the same purpose

Reply:
Something is definitely off, the phandle did not fix the build issues according to twister but locally I don't not have build errors, but the tests do not pass, despite the correct number of channels being passed.

Reply:
I also do not like the SDK Macro here and we are now using several ideas to do the same thing in this driver macro define, and there may be a bigger picture to the SDK macro that I am not aware of, but we should discuss a single way to set the size of these arrays.

Reply:
I tested the changes on the PR, no build error but again the test suite fails.

Reply:
Build errors resolved